### PR TITLE
Feat: Liquidity pools inter domain tranche token transfer

### PR DESCRIPTION
**Pull request by mustermeiszer**
# Description
The PR allows to send tranche tokens from `Domain A` to `Domain B` by relaying the message over `Centrifuge`. The Solidity side of the LP logic already is capable of doing so, this enables this feature on the chain itself.

NOTE: 
Pushing this change before the RU as the DYF pool will most likely need that feature.

## Changes and Descriptions
* Adds a new `DomainAccountToDomainAddress` converter to ease conversion from message to type
* Add different handling logic for tranche token transfers
  * Always transfer from `DomainAccount` to `T::DomainAddressToAccountId::convert(receiver)`
     &rarr; for local receiver this will just be the normal account they choose to send to
     &rarr; for remote receiver this will be a derived account that temporarily holds the tokens
  * If centrifuge is not the final receiving domain &rarr; use normal local transfer logic of LP to forward to actual domain

# Checklist:

- [x] I have added Rust doc comments to structs, enums, traits and functions
- [x] I have made corresponding changes to the documentation
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by mustermeiszer** [comment]
Should we start adding test to the pallet itself now? ^^ Integration test will be hard until the LP integration test is not done.

File: runtime/altair/src/liquidity_pools.rs [hunk 1]
```diff
@@ -1,99 +0,0 @@
-// Copyright 2023 Centrifuge Foundation (centrifuge.io).
```
Comment (lemunozm) [inline runtime/altair/src/liquidity_pools.rs:1]: Why are these pallets removed?

Comment (mustermeiszer) [reply runtime/altair/src/liquidity_pools.rs:1]: unused config, right?

File: pallets/liquidity-pools/src/inbound.rs [hunk 2]
```diff
@@ -78,12 +81,25 @@ where
...
+		// to the right domain from the holdings of the receiver we just transferred
+		// them to.
+		if receiver.domain() != Domain::Centrifuge {
+			Pallet::<T>::transfer_tranche_tokens(
+				T::RuntimeOrigin::signed(local_representation_of_receiver),
+				pool_id,
+				tranche_id,
+				receiver,
+				amount,
+			)?;
```
Comment (wischli) [inline pallets/liquidity-pools/src/inbound.rs:100]: Love the simplicity of this feature!

**Review by wischli — COMMENTED** [review]
Code LGTM! We could add a simple integration test which ensures the 3 hop of tranche token amounts via events.

File: runtime/altair/src/liquidity_pools.rs (hunk 1 above)
Comment (lemunozm) [reply runtime/altair/src/liquidity_pools.rs:1]: Ah, that's right! I forgot to remove those files after migrating them 🤦🏻‍♂️. Thanks

**Review by lemunozm — DISMISSED** [review]
LGTM!

**Review by wischli — APPROVED** [review]
I am fine with adding tests later and verifying this change on Demo. Code LGTM!

**Pull request merged**